### ISSUE TITLE
Some doubts about the code

### ISSUE THREAD

**Issue by Jiawei0o0** [ErikStammes/EADER#1]
Thanks for your job and code. 
`if labels is None:
                if self.training:
                    raise Exception('Labels are necessary during training to create Grad-CAMs')
                else:
                    labels = logits`
The code is in the attention_map_wrapper.py, from line 79 to 83.
In the testing phase, the labels are equal to logits. In my intuition, it will be something like [0.1, 0.7, 0.5...], which means it is hardly equal to zero. So, in line 88 `label_indices = labels.nonzero()`, which means that you just use all the class to generate CAM in the testing phase?
Waiting for your reply. Thanks again!

**Comment by ErikStammes** [comment]
Hi @VoooWei,
I used this code in a multi-stage pipeline, where it is common to use the ground truth labels for both the training and validation set. In a later stage I train a fully-supervised model on the proxy labels generated by this code. That model then does the inference, hence I don't need the labels for the test set. 

I hope that makes sense. Let me know if you have any other questions about the code!

**Comment by Jiawei0o0** [comment]
Thanks for your reply!
I am sorry that I forgot your task here. I had seen this strategy (Only the attention maps for the ground truth classes are kept, in your paper Soft masks.) in other papers, which choose the ground truth CAM for further classification tasks.  So I am wondering how it works in the testing phase when it faces the multi-label dataset.

**Comment by ErikStammes** [comment]
I'm not sure if I understand your question. As I said above, I have never used the model in this repository in the testing phase. I use the generated segmentation masks from the training and validation set to train another model, (e.g. PSA and then DeepLab) and use that model to generate segmentation masks on the test set. Does that answer your question?
